builtin/credential/jwt/path_cel_role.go

```diff
@@ -0,0 +1,485 @@
+		return nil, err
+	}
+	if resp.IsError() {
+		return resp, nil
+	}
+
+	// Store it
+	jsonEntry, err := logical.StorageEntryJSON("cel/role/"+roleName, entry)
+	if err != nil {
+		return nil, err
+	}
+	if err := req.Storage.Put(ctx, jsonEntry); err != nil {
+		return nil, err
+	}
+
+	if err := logical.EndTxStorage(ctx, req); err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func (b *jwtAuthBackend) pathCelRoleDelete(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	err := req.Storage.Delete(ctx, "cel/role/"+data.Get("name").(string))
+	return nil, err
+}
+
+func (b *jwtAuthBackend) getCelRole(ctx context.Context, s logical.Storage, roleName string) (*celRoleEntry, error) {
+	entry, err := s.Get(ctx, "cel/role/"+roleName)
+	if err != nil || entry == nil {
+		return nil, err
+	}
+
+	var result celRoleEntry
+	if err := entry.DecodeJSON(&result); err != nil {
+		return nil, err
+	}
+
+	result.Name = roleName
+
+	return &result, nil
+}
+
+func validateCelRoleCreation(b *jwtAuthBackend, entry *celRoleEntry, ctx context.Context, s logical.Storage) (*logical.Response, error) {
+	resp := &logical.Response{}
+
+	_, err := b.validateCelProgram(entry.CelProgram)
+	if err != nil {
+		return nil, fmt.Errorf("%w", err)
+	}
+
+	resp.Data = entry.ToResponseData()
+	return resp, nil
+}
+
+func (b *jwtAuthBackend) validateCelProgram(program celhelper.CelProgram) (bool, error) {
+	// adding a minimal jwtClaims collection here, for validating usages in CEL expression
+	_, err := b.celEvalProgram(program, map[string]any{"sub": "email@example.com", "aud": "audience", "iss": "issuer"})
+	if err != nil {
+		return false, fmt.Errorf("failed to validate CEL program: %w", err)
+	}
+	return true, nil
+}
+
+func (b *jwtAuthBackend) celEvalProgram(program celhelper.CelProgram, jwtClaims map[string]any) (any, error) {
+	env, err := b.celEnv(program)
+	if err != nil {
+		return nil, err
+	}
+	// Initialize the evaluation context for CEL expressions with the raw request data.
+	// The "request" key allows CEL expressions to access and evaluate against input fields.
+	// Additional variables and evaluated results will be added dynamically during processing.
+	evaluationData := map[string]interface{}{
+		"claims": jwtClaims,
+		"now":    time.Now(),
+	}
+
+	// Evaluate all variables
+	for _, variable := range program.Variables {
+		result, err := celhelper.ParseCompileAndEvaluateVariable(env, variable, evaluationData)
+		if err != nil {
+			return nil, fmt.Errorf("%w", err)
+		}
+
+		// Add the evaluated result for subsequent CEL evaluations.
+		// This ensures variables can reference each other and build a cumulative evaluation context.
+		evaluationData[variable.Name] = result.Value()
+	}
+
+	// Evaluate the CEL Role success expression
+	return celhelper.ParseCompileAndEvaluateExpression(env, program.Expression, evaluationData)
+}
+
+func (b *jwtAuthBackend) celEnv(program celhelper.CelProgram) (*cel.Env, error) {
```

Comment:
In the future, not for this PR as I'd like to see it merged, but we can add `EvaluateRole(...)` as a helper, so the caller can have easy enforcement.